codec-http2/src/main/java/io/netty/handler/codec/http2/Http2LocalFlowController.java

```diff
@@ -66,4 +66,50 @@ void receiveFlowControlledFrame(ChannelHandlerContext ctx, Http2Stream stream, B
      * @return the number of unconsumed bytes for the stream.
      */
     int unconsumedBytes(Http2Stream stream);
+
+    /**
+     * Factory method to generate a new {@link LocalFlowState} object. This will be called after a new stream object
+     * is created and before {@link Http2Connection.Listener#onStreamAdded(Http2Stream)} is called. The result of this
+     * method will be returned by {@link Http2Stream#localFlowState()} for use by {@link Http2LocalFlowController}.
+     * @param stream The stream to associate the {@link LocalFlowState} with.
+     * This may be any stream, including the {@link Http2Connection#connectionStream()}.
+     * @return A new object which will represent local flow control state for {@code stream}.
+     */
+    LocalFlowState newLocalFlowState(Http2Stream stream);
```

Comment:
If we are going to have a common interface for the two I think we need to evaluate if we can define a meaningful intersection of the 2 interfaces.  Seems like we should be able to do this because at the end of the day they are both just counting bytes consumed, but I think we moved away from this in the past.  What about something like the following:

```java
public interface FlowControlState {
  // The amount of bytes that can be used either to send/receive flow controlled frame payload
  int availableWindow();

  // the amount of bytes that is allocated for flow control but not yet returned to flow control.
  int unavailableWindow();
}
```

This interface may belong in the `Http2Stream` class as it is specific to a stream?